crates/router/src/connector/nexinets/transformers.rs

```diff
@@ -525,13 +586,79 @@ fn get_payment_details_and_product(
     }
 }
 
-fn get_card_details(req_card: &api_models::payments::Card) -> NexinetsPaymentDetails {
-    NexinetsPaymentDetails::Card(Box::new(NexiCardDetails {
+fn get_card_data(
+    item: &types::PaymentsAuthorizeRouterData,
+    card: &api_models::payments::Card,
+) -> Result<NexinetsPaymentDetails, errors::ConnectorError> {
+    if item.request.is_mandate_payment() {
```

Comment:
replace this with below code 
```suggestion
let (card_data, cof_contract) = match item.request.is_mandate_payment() {
        true => {
            let card_data = match item.request.off_session {
                Some(true) => CardDataDetails::PaymentInstrument(Box::new(PaymentInstrument {
                    payment_instrument_id: item.request.connector_mandate_id(),
                })),
                _ => CardDataDetails::CardDetails(Box::new(get_card_details(card))),
            };
            let cof_contract = Some(CofContract {
                recurring_type: RecurringType::Unscheduled,
            });
            (card_data, cof_contract)
        }
        false => (
            CardDataDetails::CardDetails(Box::new(get_card_details(card))),
            None,
        ),
    };
    Ok(NexinetsPaymentDetails::Card(Box::new(NexiCardDetails {
        card_data,
        cof_contract,
    })))
```

Lets not have early returns